### ISSUE TITLE
ERROR TypeError: "this.container is undefined" for NbDialog

### ISSUE THREAD

**Issue by beyondszine** [akveo/nebular#2060]
Issue type

I'm submitting a 
* [ ] bug report
* [ ] feature request
* Unknown cause, Not qualified enough to call this a bug, & certainly not a feature request.  Just that I am not able to get NbDialog working because of this not much descriptive error & my incapability to understand the complicated stack trace.

Issue description
NbDialog service is not able to open a dialog box by giving an error " ERROR TypeError: "this.container is undefined" for NbDialog"

Steps to reproduce:
https://angular-acregw.stackblitz.io

Kindly help by giving it a look.

**Comment by caiomcoliveira** [comment]
Couldn't open the stackblitz

Did you put the component you are trying to open on EntryComponents?

**Comment by maihannijat** [comment]
I think it's because you are not using Nebular Layout. Put your content inside `<nb-layout></nb-layout>`

**Comment by beyondszine** [comment]
> Couldn't open the stackblitz
> 
> Did you put the component you are trying to open on EntryComponents?

@caiomcoliveira yes indeed.  I am not sure why stackblitz couldn't open.  Can you please retry via 
https://stackblitz.com/edit/angular-acregw
Yes, it is in the entryComponents list.

**Comment by beyondszine** [comment]
> I think it's because you are not using Nebular Layout. Put your content inside `<nb-layout></nb-layout>` in the app.component.html
> 
> And consider the @caiomcoliveira, otherwise it would throw:
> 
> > error: No component factory found for ArtefactdetailComponent. Did you add it to @NgModule.entryComponents?

Indeed, the html code inside app.component.html is wrapped under <nb-layout>  but yet the error is same.

**Comment by maihannijat** [comment]
@beyondszine The following is the correct way of wrapping in nb-layout:
`template : '<nb-layout><nb-layout-column><button nbButton (click)="open()">Open Dialog</button></nb-layout-column></nb-layout>'`

**Comment by beyondszine** [comment]
Thanks @maihannijat 
This way, it works for sure.
I have been appalled by the browsers error display on console.  Firefox says: 
![todelete](https://user-images.githubusercontent.com/2175509/68316596-b9045f80-00df-11ea-9291-67532a52843c.png)

while Chrome showed an error of 
![todelete_chrome](https://user-images.githubusercontent.com/2175509/68316703-e6510d80-00df-11ea-914d-c411e584771c.png)

- Just adding \<nb-layout\> tag without \<nb-layout-column\> makes all the html hide, no error message or anything on the console.  Just nothing appears on screen for both the browsers.

Regarding the documentation & guides I have been following:
- https://akveo.github.io/nebular/docs/components/dialog/examples#nbdialogservice 
- https://github.com/akveo/nebular/blob/master/src/playground/with-layout/dialog/components/showcase-dialog.component.ts

For all of the files( Loader Component, Module, & the component to be loaded), I wonder how things are working here. 
Am I still overlooking something crucial, or the docs & guides need update?

**Comment by yggg** [comment]
@beyondszine Can't open stackblitz. Make sure it's not private.
![image](https://user-images.githubusercontent.com/6151971/69342476-b461ae80-0c7c-11ea-9962-2e64bb26cdb5.png)

**Comment by MrTaiji** [comment]
I Actually got the same Error (with both kinds of Error Codes in Firefox and Chrome), which caused the whole Application to break. After checking the error Code in Chrome and more Debugging I found out that adding the NbTheme Module Import in any submodule, conflicts with all overlays from Angular Material (matDialog, matSnackbar, matMenu) 

See my stackoverflow post here: https://stackoverflow.com/questions/63849598/angular-nebular-chat-only-for-one-module-and-component
